linux: do not initialize dbus/systemd-logind if it's disabled.

How to reproduce the issue:

- system has booted with no systemd and dbus has been started.
- system contains systemd installed (dbus services, libs, etc)
- InitOutput() thinks that systemd is running and issues dbus requests
for logind, resulting in an error like this:

/dev/dri/card0: failed to set DRM interface version 1.4: Permission denied

See https://github.com/voidlinux/void-packages/issues/532
for more information.

--- hw/xfree86/common/xf86Init.c.orig	2014-09-04 16:18:20.866121037 +0200
+++ hw/xfree86/common/xf86Init.c	2014-09-04 16:19:28.401962814 +0200
@@ -463,8 +463,10 @@ InitOutput(ScreenInfo * pScreenInfo, int
         if (xf86DoShowOptions)
             DoShowOptions();
 
+#ifdef SYSTEMD_LOGIND
         dbus_core_init();
         systemd_logind_init();
+#endif
 
         /* Do a general bus probe.  This will be a PCI probe for x86 platforms */
         xf86BusProbe();
